src/pdm/utils.py

```diff
@@ -413,8 +413,8 @@ def deprecation_warning(message: str, stacklevel: int = 1, raise_since: str | No
 
     if raise_since is not None:
         if Version(__version__) >= Version(raise_since):
-            raise DeprecationWarning(message)
-    warnings.warn(message, DeprecationWarning, stacklevel=stacklevel + 1)
+            raise FutureWarning(message)
+    warnings.warn(message, FutureWarning, stacklevel=stacklevel + 1)
```

Comment:
Why future warnings instead of deprecation warnings? The function  is called `deprecation_warning`. The test function as well.

Reply:
Found this: https://stackoverflow.com/questions/55377831/difference-between-deprecationwarning-pendingdeprecationwarning-and-futurewarni

TIL

Reply:
@pawamoy This was motivated in the commit message.

Reply:
@sanmai-NL I missed your commit message indeed, my bad. I agree with not changing the function name as to avoid a breaking change.